### PR TITLE
GH-45 - Integrate GolangCI-lint

### LINKED ISSUE #45: Integrate GolangCI-Lint
Mattermost is currently in the process of using [`golangci-lint`](https://github.com/golangci/golangci-lint/) as the default linter for all go projects. https://github.com/mattermost/mattermost-plugin-autolink/pull/108 serves as an example of how the migration should look like:
1. Copy `.golangci.yml` from https://github.com/mattermost/mattermost-plugin-autolink/ into this repository.
2. Copy `Makefile` from https://github.com/mattermost/mattermost-plugin-autolink/ into this repository.
3. Copy `build/` from https://github.com/mattermost/mattermost-plugin-autolink/ into this repository.
4. Run `go mod tidy` and `make dist` to ensure the build system works fine.
5. Update `linters-settings.goimports.local-prefixes` to the go module path i.e. to `github.com/mattermost/mattermost-plugin-welcomebot`.
6. Run `make golangci-lint` and fix all issues that arise.

Fixing an linter issue is sometimes trivial but can also be quite complicated. There might be cases, where ignoring an issue using [`issues.exclude-rules`](https://github.com/golangci/golangci-lint/#config-file) is needed. If you don't know how to resolve an error, please reach out to `@hanzei` on the [Mattermost Community Server](https://community.mattermost.com) or ask the question here.

### PR DESCRIPTION
**Pull request by nathanaelhoun**
#### Summary

This PR copies most of the golangci-lint configuration from mattermost/mattermost-plugin-starter-template#90 into this repo. This helps with a consistent code style and should improve code quality over time.

Most changes are straightforward and just style wise. 

Inspired by mattermost/mattermost-plugin-autolink#108.

#### Ticket Link

Closes #45

### CODE REVIEW

File: server/welcomebot.go [hunk 1]
```diff
@@ -135,7 +135,7 @@ func (p *Plugin) renderWelcomeMessage(messageTemplate MessageTemplate, configMes
 
 	tmpMsg, _ := template.New("Response").Parse(strings.Join(configMessage.Message, "\n"))
 	var message bytes.Buffer
-	tmpMsg.Execute(&message, messageTemplate)
+	_ = tmpMsg.Execute(&message, messageTemplate)
```
Comment (hanzei) [inline server/welcomebot.go:138]: Logging these errors is valuable, as it can greatly help with debugging

Comment (nathanaelhoun) [reply server/welcomebot.go]: I have added a simple log of this error in [`685000`](https://github.com/mattermost/mattermost-plugin-welcomebot/pull/51/commits/685000ca2f2eda5812ce7ef2b09451a69774816e). Is this sufficient ?

File: server/welcomebot.go [hunk 2]
```diff
@@ -145,7 +151,7 @@ func (p *Plugin) renderWelcomeMessage(messageTemplate MessageTemplate, configMes
 	if len(configMessage.AttachmentMessage) > 0 || len(actionButtons) > 0 {
 		tmpAtch, _ := template.New("AttachmentResponse").Parse(strings.Join(configMessage.AttachmentMessage, "\n"))
 		var attachMessage bytes.Buffer
-		tmpAtch.Execute(&attachMessage, messageTemplate)
+		_ = tmpAtch.Execute(&attachMessage, messageTemplate)
```
Comment (hanzei) [inline server/welcomebot.go:154]: Also here

File: server/welcomebot.go [hunk 3]
```diff
@@ -192,7 +198,7 @@ func (p *Plugin) processActionMessage(messageTemplate MessageTemplate, action *A
 
 	tmpMsg, _ := template.New("Response").Parse(strings.Join(configMessageAction.ActionSuccessfulMessage, "\n"))
 	var message bytes.Buffer
-	tmpMsg.Execute(&message, messageTemplate)
+	_ = tmpMsg.Execute(&message, messageTemplate)
```
Comment (hanzei) [inline server/welcomebot.go:201]: And here

Comment (nathanaelhoun) [reply server/welcomebot.go]: Ofc, I should have thought about it.Done in [`166ff26`](https://github.com/mattermost/mattermost-plugin-welcomebot/pull/51/commits/166ff263682e7619e2859f9436c829a38de9698c)

**Comment by hanzei** [comment]
Awesome, thanks @nathanaelhoun :rocket:

**Pull request merged**